App.js

```diff
@@ -78,6 +90,19 @@ class App extends React.Component {
     this.initialRoute = userStore.authenticated ? 'Main' : 'Auth';
   };
 
+  listenForNotifications = () => {
+    Notifications.addListener(notification => {
+      if (notification.origin === 'received' && Platform.OS === 'ios') {
+        ReactAlert.alert(notification.title, notification.body);
+      }
+    });
+  };
+
+  componentWillMount() {
+    getiOSNotificationPermission();
+    this.listenForNotifications();
```

Comment:
It already asks for the notification if the user is still logged out. Please check here if the user is logged in.